cameo/api/designer.py

```diff
@@ -197,28 +205,31 @@ def optimize_strains(self, pathways, view, aerobic=True):
             A data frame with strain designs processed and ranked.
 
         """
-        opt_gene_runner = _OptGeneRunner(self.debug)
-        differential_fva_runner = _DifferentialFVARunner(self.debug)
-        strategies = [(host, model, pathway, aerobic) for (host, model) in pathways for pathway in pathways[host, model]
-                      if pathway.needs_optimization(model, objective=model.biomass)]
+        strategies = [{'host': host.name, 'model': model, 'pathway': pathway, 'aerobic': aerobic} for (host, model) in
+                      pathways for pathway in pathways[host, model] if
+                      pathway.needs_optimization(model, objective=model.biomass)]
 
         print("Optimizing %i pathways" % len(strategies))
 
         results = DataFrame(columns=["host", "model", "manipulations", "heterologous_pathway",
                                      "fitness", "yield", "product", "biomass", "method"])
 
-        mapped_designs1 = view.map(opt_gene_runner, strategies)
-        mapped_designs2 = view.map(differential_fva_runner, strategies)
-        progress = ProgressBar(maxval=len(mapped_designs1) + len(mapped_designs2),
```

Comment:
This is only for the processing of the results, which really doesn't take that long so I don't see the point of having a progress bar.